### PR TITLE
React to some HTTP responses

### PR DESCRIPTION
**Pull request by MaciejG604**
We can point out possible reasons of error HTTP responses according to: https://central.sonatype.org/faq/

### CODE REVIEW

File: modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala [hunk 1]
```diff
@@ -1071,6 +1080,50 @@ object Publish extends ScalaCommand[PublishOptions] with BuildCommandHelpers {
...
+            case (_, _, _: Upload.Error.Unauthorized) => true
+            case _                                    => false
+          } =>
+        logger.error(
+          s"""
+             |Uploading files failed!
+             |Possible causes:
+             |- incorrect Sonatype credentials
+             |- your Sonatype password or username may contain unsupported characters
+             |- incorrect Sonatype server was used (legacy or s01)
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala]: we should be able to check which server was used here, I think?

File: modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala [hunk 2]
```diff
@@ -1071,6 +1080,50 @@ object Publish extends ScalaCommand[PublishOptions] with BuildCommandHelpers {
...
+      case _ :: _ if isSonatype && errors.forall {
+            case (_, _, _: Upload.Error.Unauthorized) => true
+            case _                                    => false
+          } =>
+        logger.error(
+          s"""
+             |Uploading files failed!
+             |Possible causes:
+             |- incorrect Sonatype credentials
+             |- your Sonatype password or username may contain unsupported characters
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala]: can we check earlier in our logic if any characters unsupported by `Sonatype` were used in the credentials?

**Comment by MaciejG604** [comment]
TEST 123

**Comment by MaciejG604** [comment]
Suggested Reviewers:
1. @tgodzik
2. MaciejG604 
3. keynmol

**Comment by MaciejG604** [comment]
Suggested Reviewers:
1. @tgodzik
2. @Gedochao 
3. @keynmol

File: modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala [hunk 3]
```diff
@@ -56,13 +56,7 @@ import scala.cli.commands.shared.{
...
 import scala.cli.commands.{ScalaCommand, SpecificationLevel, WatchUtil}
 import scala.cli.config.{ConfigDb, Keys, PasswordOption, PublishCredentials}
-import scala.cli.errors.{
-  FailedToSignFileError,
-  MalformedChecksumsError,
-  MissingConfigEntryError,
-  MissingPublishOptionError,
-  UploadError
-}
+import scala.cli.errors._
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/commands/publish/Publish.scala]: NIT might be overkill to import all errors here. Also, use `*` syntax, if you leave it, `_` is to be deprecated soon

File: modules/cli/src/main/scala/scala/cli/errors/InvalidSonatypePublishCredentials.scala [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package scala.cli.errors
+
+import scala.build.errors.BuildException
+
+final class InvalidSonatypePublishCredentials(usernameIsAscii: Boolean, passwordIsAscii: Boolean)
+    extends BuildException(
+      if (usernameIsAscii && passwordIsAscii)
+        "Username and password to the publish repository are incorrect"
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/errors/InvalidSonatypePublishCredentials.scala]: NIT
```suggestion
        "Username or password to the publish repository are incorrect"
```

**Review by Gedochao — APPROVED** [review]
LGTM, left some NIT comments

**Pull request merged**